mdstat.go

```diff
@@ -42,75 +44,82 @@ type MDStat struct {
 	BlocksSynced int64
 }
 
-// ParseMDStat parses an mdstat-file and returns a struct with the relevant infos.
-// TODO: this should just be called "MDStat" and use a separate parseMDStat function
-func (fs FS) ParseMDStat() (mdstates []MDStat, err error) {
-	mdStatusFilePath := fs.proc.Path("mdstat")
-	content, err := ioutil.ReadFile(mdStatusFilePath)
+// MDStat parses an mdstat-file (/proc/mdstat) and returns a slice of
+// structs containing the relevant info.  More information available here:
+// https://raid.wiki.kernel.org/index.php/Mdstat
+func (fs FS) MDStat() ([]MDStat, error) {
+	f, err := os.Open(fs.proc.Path("mdstat"))
 	if err != nil {
-		return []MDStat{}, fmt.Errorf("error parsing %s: %s", mdStatusFilePath, err)
+		return nil, err
 	}
+	defer f.Close()
 
-	mdStates := []MDStat{}
-	lines := strings.Split(string(content), "\n")
-	for i, l := range lines {
-		if l == "" {
-			continue
-		}
-		if l[0] == ' ' {
-			continue
-		}
-		if strings.HasPrefix(l, "Personalities") || strings.HasPrefix(l, "unused") {
+	return parseMDStat(f)
+}
+
+// parseMDStat parses an io.Reader (/proc/mdstat) and returns a slice of
+// structs containing the relevant info.
+func parseMDStat(r io.Reader) (mdStats []MDStat, err error) {
+	mdStats = []MDStat{}
+	s := bufio.NewScanner(r)
+	for s.Scan() {
+		line := s.Text()
+		// Ignore these lines
+		if strings.TrimSpace(line) == "" ||
+			strings.HasPrefix(line, "Personalities") ||
+			strings.HasPrefix(line, "unused") {
 			continue
 		}
 
-		mainLine := strings.Split(l, " ")
-		if len(mainLine) < 3 {
-			return mdStates, fmt.Errorf("error parsing mdline: %s", l)
+		mdStat := MDStat{}
```

Comment:
Since this is referred to many times and the function is about parsing these structures, consider a short name like `m`.

Reply:
Done